Add CentOS7 dockerfile

discogestalt opened the pull request:
Add support for rpm building to CentOS 8 dockerfile

discogestalt committed:
Add CentOS7 dockerfile

Add support for rpm building to CentOS 8 dockerfile

tomerd commented on swift-ci/master/centos/8/Dockerfile:
```diff
@@ -30,6 +30,7 @@ RUN yum install --enablerepo=PowerTools -y \
     python3-six           \
     python3-pexpect       \
     platform-python-devel \
+    rpm-build.x86_64      \
```
hi @discogestalt we should remove  rpm-build package from these docker images - we do not intend to create RPMs using these docker images want to keep the list minimal

discogestalt replied:
Ok, revision coming up.

tomerd commented on swift-ci/master/centos/7/Dockerfile:
```diff
@@ -0,0 +1,54 @@
+	sqlite-devel      \
+	swig3             \
+	which             \
+	zlib-devel
```
@discogestalt this list is significantly longer than what we do for Centos8 and other distributions. Is it the bare minimum to get Swift build & qualified?

discogestalt replied:
For the most part, yeah - let me confirm, there may be a package or two that were there for building libcurl (when that was necessary).

discogestalt committed:
Clean up package install list

Put EL7 lib fixes in the docker file.

Remove RPM build lines from CentOS8 docker file.
